### PR TITLE
Update Parent POM + facelift

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Just an update towards the next release.

- [x] - Update Parent POM to 1.39
- [x] - Fix Upper Bounds issues
- [x] - Fix issues reported by FindBugs

Snapshot: 1.26-20171006.214254-1

Downstream PRs:
* https://github.com/jenkinsci/envinject-api-plugin/pull/3
* https://github.com/jenkinsci/envinject-plugin/pull/126

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@reviewbybees @dwnusbaum

File: src/main/java/org/jenkinsci/lib/envinject/EnvInjectAction.java [hunk 1]
```diff
@@ -166,11 +166,6 @@ public String transformEntry(String key, String value) {
             return null;
         }
 
-        Job<?, ?> project = build.getParent();
```
Comment (dwnusbaum) [inline src/main/java/org/jenkinsci/lib/envinject/EnvInjectAction.java]: Is `build.getParent` guaranteed to be non-null? If so the condition in `getEnvVars` shouldn't check if it is null.

Comment (oleg-nenashev) [reply src/main/java/org/jenkinsci/lib/envinject/EnvInjectAction.java]: Javadoc says so...
If you have any concerns, I can revert this change and suppress FindBugs. This is a deprecated code anyway

Comment (dwnusbaum) [reply src/main/java/org/jenkinsci/lib/envinject/EnvInjectAction.java]: No this is good, I just mean that #11 added code that checks if it is null, so can you remove that check here?

**Comment by dwnusbaum** [comment]
🐝

**Comment by oleg-nenashev** [comment]
@reviewbybees done
🚢 🇮🇹

**Pull request merged**